openmc/deplete/abc.py

```diff
@@ -160,3 +163,165 @@ def get_results_info(self):
 
     def finalize(self):
         pass
+
+
+class ReactionRateHelper(ABC):
+    """Abstract class for generating reaction rates for operators
+
+    Responsible for generating reaction rate tallies for burnable
+    materials, given nuclides and scores from the operator.
+
+    Reaction rates are passed back to the operator for be used in
+    an :class:`openmc.deplete.OperatorResult` instance
+
+    Attributes
+    ----------
+    nuclides : list of str
+        All nuclides with desired reaction rates. Ordered to be
+        consistent with :class:`openmc.deplete.Operator`
+    """
+
+    def __init__(self):
+        self._nuclides = None
+        self._rate_tally = None
+        self._results_cache = None
+
+    @abstractmethod
+    def generate_tallies(self, materials, scores):
+        """Use the capi to build tallies needed for reaction rates"""
+
+    @property
+    def nuclides(self):
+        """List of nuclides with requested reaction rates"""
+        return self._nuclides
+
+    @nuclides.setter
+    def nuclides(self, nuclides):
+        check_type("nuclides", nuclides, list, str)
+        self._nuclides = nuclides
+        self._rate_tally.nuclides = nuclides
+
+    def _reset_results_cache(self, nnucs, nreact):
```

Comment:
I find it a little weird that 1) this method is responsible for both creating and resetting the cache and 2) it also returns the cache. To me, the name `_reset_results_cache` would indicate that there's no return value. That being said, I don't have any concrete suggestions for improving this :man_shrugging:

Reply:
Fair point. I've renamed the method to be `_get_results_cache` and then filled with zeros during the `get_material_rates` method.